src/Control/Monad/Trans/MSF/Maybe.hs

```diff
@@ -65,11 +66,14 @@ catchMaybe msf1 msf2 = safely $ do
 
 -- * Converting to and from 'MaybeT'
 
+exceptToMaybeS :: (Functor m, Monad m) => MSF (ExceptT e m) a b -> MSF (MaybeT m) a b
+exceptToMaybeS = morphS $ MaybeT . fmap (either (const Nothing) Just) . runExceptT
```

Comment:
For info only: what goes in the fmap is:
- http://hackage.haskell.org/package/extra-1.6.18/docs/src/Data.Either.Extra.html#eitherToMaybe
- http://hackage.haskell.org/package/either-4.3.3.2/docs/src/Data-Either-Combinators.html#rightToMaybe
- It also happens to match the behaviour, but not the signature, of https://hackage.haskell.org/package/base-4.12.0.0/docs/src/Data.Maybe.html#listToMaybe. It can also be implemented as `(listToMaybe . toList)`.

It probably makes sense to have something this generic defined somewhere in base, probably in foldable.

I'll happily accept it as is, but it may be useful to know that this expression can be simplified.

Reply:
I think this doesn't match the behaviour of `listToMaybe`. `listToMaybe` forgets the tail of a list.

Reply:
I only mean `either (const Nothing) Just`.

Reply:
Ah, I see. Yes, I was tempted to use these functions from `extra` or `either`, but didn't want to add a dependency. If you prefer an additional dependency, we can do that as well.

Reply:
Nah, it's fine. It may be more appropriate to suggest to the libraries people to give this a name and put it in foldable. The definition is small enough that we can just add it, even if there's a shorter way or is defined in some library (just one small definition does not warrant adding a dependency like that).

There is some relation between foldables, traversables and streams that is bugging me and I can't see clearly, but would make this implementation trivial.

Reply:
I don't see how a function of type `Either e a -> Maybe a` is related to `Foldable` in a meaningful way.

There is definitely a relation between `Foldable`, `Traversable` and `MSF Maybe () a`. A `Foldable` is simply any list structure (without preferring a particular shape like linked lists or balanced trees over others). A `Traversable` is a `Foldable` that is compatible with `Functor`. I think `MSF Maybe () a` is both, because it's basically a list. Compare the signature of `accumulateWith` with `foldr`.

Reply:
> I don't see how a function of type `Either e a -> Maybe a` is related to `Foldable` in a meaningful way.

There are ways to implement it in which the type would be more general than that. For one example, the implementation in https://hackage.haskell.org/package/base-4.12.0.0/docs/src/Data.Maybe.html#listToMaybe does not mention `Either` at all.

It is too small a piece of code to argue about it. With my original message, I only wanted to point out that this is very specific and there's probably code somewhere that does this for more general types, or that this could be generally useful for other types. If the solution is not immediately obvious, we don't need to act on this and I don't want to open an issue about it. It'll come up again as we clean the code more and more.